image/oci/test/test_oci_manifest.py

```diff
@@ -0,0 +1,168 @@
+        "sha256:3c3a4604a545cdc127456d94e421cd355bca5b528f4a9c1905b15da2eb4a4c6b",
+        "sha256:ec4b8955958665577945c89419d1af06b5f7636b4ac3da7f12184802ad867736",
+        "sha256:b5b2b2c507a0944348e0303114d8d93aaaa081732b86451d9bce1f432a537bc7",
+    ]
+
+    assert manifest.local_blob_digests == manifest.blob_digests
+
+    assert len(manifest.filesystem_layers) == 3
+    assert (
+        str(manifest.leaf_filesystem_layer.digest)
+        == "sha256:ec4b8955958665577945c89419d1af06b5f7636b4ac3da7f12184802ad867736"
+    )
+
+    assert not manifest.has_remote_layer
+    assert manifest.has_legacy_image
+    assert manifest.annotations == {u"com.example.key1": u"value1", u"com.example.key2": u"value2"}
+
+
+def test_parse_basic_remote_manifest():
+    manifest = OCIManifest(Bytes.for_string_or_unicode(SAMPLE_REMOTE_MANIFEST))
+    assert not manifest.is_manifest_list
+    assert (
+        manifest.digest == "sha256:dd18ed87a00474aff683cee7160771e043f1f0eadd780232715bc0678a984a5e"
+    )
+
+    assert manifest.blob_digests == [
+        "sha256:9834876dcfb05cb167a5c24953eba58c4ac89b1adf57f28f2f9d09af107ee8f0",
+        "sha256:3c3a4604a545cdc127456d94e421cd355bca5b528f4a9c1905b15da2eb4a4c6b",
+        "sha256:ec4b8955958665577945c89419d1af06b5f7636b4ac3da7f12184802ad867736",
+        "sha256:b5b2b2c507a0944348e0303114d8d93aaaa081732b86451d9bce1f432a537bc7",
+    ]
+
+    assert manifest.local_blob_digests == [
+        "sha256:9834876dcfb05cb167a5c24953eba58c4ac89b1adf57f28f2f9d09af107ee8f0",
+        "sha256:ec4b8955958665577945c89419d1af06b5f7636b4ac3da7f12184802ad867736",
+        "sha256:b5b2b2c507a0944348e0303114d8d93aaaa081732b86451d9bce1f432a537bc7",
+    ]
+
+    assert len(manifest.filesystem_layers) == 3
+    assert (
+        str(manifest.leaf_filesystem_layer.digest)
+        == "sha256:ec4b8955958665577945c89419d1af06b5f7636b4ac3da7f12184802ad867736"
+    )
+
+    assert manifest.has_remote_layer
+
+    assert not manifest.has_legacy_image
+    assert not manifest.get_legacy_image_ids(None)
+
+
+def test_get_schema1_manifest():
+    retriever = ContentRetrieverForTesting.for_config(
+        {
+            "config": {"Labels": {"foo": "bar",},},
+            "rootfs": {"type": "layers", "diff_ids": []},
+            "history": [
+                {"created": "2018-04-03T18:37:09.284840891Z", "created_by": "foo"},
+                {"created": "2018-04-12T18:37:09.284840891Z", "created_by": "bar"},
+                {"created": "2018-04-03T18:37:09.284840891Z", "created_by": "foo"},
+            ],
+            "architecture": "amd64",
+            "os": "linux",
+        },
+        "sha256:b5b2b2c507a0944348e0303114d8d93aaaa081732b86451d9bce1f432a537bc7",
+        7023,
+    )
+
+    manifest = OCIManifest(Bytes.for_string_or_unicode(SAMPLE_MANIFEST))
+    assert manifest.get_manifest_labels(retriever) == {
+        u"com.example.key1": u"value1",
+        u"com.example.key2": u"value2",
+        "foo": "bar",
```

Comment:
Why are the first two unicode while the this one bytes?

Reply:
Can we remove the `u` prefixes then. It kinda suggests that the first two labels are parsed differently than the last.

Reply:
Sure